pandas/tests/groupby/test_rank.py

```diff
@@ -290,3 +290,15 @@ def test_rank_empty_group():
     result = df.groupby(column).rank(pct=True)
     expected = DataFrame({"B": [0.5, np.nan, 1.0]})
     tm.assert_frame_equal(result, expected)
+
+
+def test_rank_zero_div():
+    # GH 23666
+    df = DataFrame({
```

Comment:
can you add a case which has a NaN, one which has multiple values in a group, and one which has NO values in the group.